core/include/moveit/task_constructor/task.h

```diff
@@ -126,9 +126,10 @@ class Task : protected WrapperBase
 	void init();
 
 	/// reset, init scene (if not yet done), and init all stages, then start planning
+	/// when preempting a plan, you must reset the preempt flag before subsequent planning
 	moveit::core::MoveItErrorCode plan(size_t max_solutions = 0);
-	/// interrupt current planning (or execution)
-	void preempt();
+	/// interrupt current planning (or execution) or reset the preempt flag (false)
```

Comment:
Execution cannot be preempted with this call.
```suggestion
	/// interrupt current planning or reset the preempt flag (false)
```

Reply:
That would be more clean. We would need a try catch block and catch everything in the plan() to reset the value.